### ISSUE TITLE
Provide a way to specify licenses for each contained file.

### ISSUE THREAD

**Issue by pjf** [KSP-CKAN/CKAN#22]
Some mods will have multiple licenses for different assets (eg, code with one license, graphics another).  The [debian spec](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#files-paragraph) already has a way of managing this, we should do the same.

Luckily, the Debian spec makes it really easy, because it uses file patterns. Thus `*.png *.jpg` or `images/*` provide easy ways to tag a number of files at once.

Many thanks to @eggrobin for raising this. <3

**Comment by politas** [comment]
Not mentioned above is the point that Mod authors are able to dual or treble-licence their code and assets. For instance, code can be released under GPL and Artistic, allowing redistributors to choose which terms they want to accept. To be good FOSS citizens, we should pass on the same freedom of choice to our users, since we are making no modifications that require us to choose between licences.
Multiple licence information should be presented to users in the CKAN client.

**Comment by pjf** [comment]
> Not mentioned above is the point that Mod authors are able to dual or treble-licence their code and assets (or more).

@politas: We've supported multiple licenses since 3a311bb1, which pre-dates the existence of this ticket, and much of the CKAN codebase.

* https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#license
* https://github.com/KSP-CKAN/CKAN/blob/master/CKAN.schema#L67-L69

These should only be used when the *whole mod* is multi-licensed. (Eg: GPL or Artistic, which is common in the Perl community.)

**Comment by pjf** [comment]
As has been pointed out on IRC, whether or not the clients do the right thing may be another matter, but the spec definitely says we support multiple licenses, and so if anything doesn't that's a bug, not a design decision.

**Comment by politas** [comment]
Yes, the schema supports multiple licences, but neither the CKAN gui client nor NetKAN support them.

**Comment by pjf** [comment]
> Yes, the schema supports multiple licences, but neither the CKAN gui client nor NetKAN support them.

Aww yis! You've found a bug! Opening a ticket now.

**Comment by pjf** [comment]
Now tracking the bug side of this in #966.

Having per-file licenses still remains a desired feature. :)

**Comment by pjf** [comment]
This ticket was almost an innocent bystander of the great merge. Re-opening. :)

**Comment by politas** [comment]
So, the current situation is that when different assets are covered by different licences, the "most restrictive" licence is presented/used. I've been looking for the discussion that lead to this policy, and I can't find it. Can anyone point me to it? I have contradictory arguments, but I don't want to just rehash previous discussions.

**Comment by pjf** [comment]
@politas : The discussion may have happened over IRC, but the guts of it is that we use (or will use) the license information to figure out if we can legally redistribute a mod. If a mod contains mixed content, then if *any* of that content forbids redistribution, that knocks out the whole zipfile. Other systems may also do the same.

Having said that, we should probably have a license string that indicates we're permitted distribution without modification. I think that RemoteTech is `restricted` (as some of the assets are non-free), but I believe redistribution of the work as a whole is permitted.

**TL;DR:** It's so our bots don't infringe.

**Comment by politas** [comment]
Ok, it sounds like this is an area that needs to be discussed in a way that is accessible. For a start, it can be difficult to assess which of multiple licences is the "most restrictive" if they are restrictive in different ways. For instance,commercial re-use clauses versus advertising clauses versus no derivatives clauses. If what we want is something that tells the bots whether they can redistribute a mod, we should have a separate field that stores that specific piece of information. It can be automatically filled based on a single licence, or it can be over-ridden to match a specific mod's multi-licence situation.
